src/autopas/baseFunctors/CellFunctor3B.h

```diff
@@ -237,402 +281,315 @@ void CellFunctor3B<Particle, ParticleCell, ParticleFunctor, DataLayout, useNewto
+        if (std::abs(p1Projection - p2Projection) > _sortingCutoff) {
           break;
         }
-        Particle &p2 = *p2Iter->second;
-        auto p3Iter = p2Iter;
-        ++p3Iter;
 
-        for (; p3Iter != cellSorted._particles.end(); ++p3Iter) {
-          if (std::abs(p2Iter->first - p3Iter->first) > _sortingCutoff ||
-              std::abs(p1Iter->first - p3Iter->first) > _sortingCutoff) {
+        for (auto cellIter3 = std::next(cellIter2); cellIter3 != cellSorted._particles.end(); ++cellIter3) {
+          auto &[p3Projection, p3Ptr] = *cellIter3;
+          if (std::abs(p1Projection - p3Projection) > _sortingCutoff or
+              std::abs(p2Projection - p3Projection) > _sortingCutoff) {
             break;
           }
-          Particle &p3 = *p3Iter->second;
-          if constexpr (newton3) {
-            _functor->AoSFunctor(p1, p2, p3, true);
-          } else {
-            _functor->AoSFunctor(p1, p2, p3, false);
-            _functor->AoSFunctor(p2, p1, p3, false);
-            _functor->AoSFunctor(p3, p1, p2, false);
-          }
+          interactParticles(*p1Ptr, *p2Ptr, *p3Ptr);
         }
       }
     }
   } else {
-    for (auto p1Iter = cell.begin(); p1Iter != cell.end(); ++p1Iter) {
-      Particle &p1 = *p1Iter;
-      auto p2Iter = p1Iter;
-      ++p2Iter;
-
-      for (; p2Iter != cell.end(); ++p2Iter) {
-        Particle &p2 = *p2Iter;
-        auto p3Iter = p2Iter;
-        ++p3Iter;
-
-        for (; p3Iter != cell.end(); ++p3Iter) {
-          Particle &p3 = *p3Iter;
-
-          if constexpr (newton3) {
-            _functor->AoSFunctor(p1, p2, p3, true);
-          } else {
-            _functor->AoSFunctor(p1, p2, p3, false);
-            _functor->AoSFunctor(p2, p1, p3, false);
-            _functor->AoSFunctor(p3, p1, p2, false);
-          }
+    for (auto p1Ptr = cell.begin(); p1Ptr != cell.end(); ++p1Ptr) {
+      auto p2Ptr = p1Ptr;
+      ++p2Ptr;
+      for (; p2Ptr != cell.end(); ++p2Ptr) {
+        auto p3Ptr = p2Ptr;
+        ++p3Ptr;
+        for (; p3Ptr != cell.end(); ++p3Ptr) {
+          interactParticles(*p1Ptr, *p2Ptr, *p3Ptr);
         }
       }
     }
   }
 }
 
-template <class Particle, class ParticleCell, class ParticleFunctor, DataLayoutOption::Value DataLayout,
-          bool useNewton3, bool bidirectional>
-void CellFunctor3B<Particle, ParticleCell, ParticleFunctor, DataLayout, useNewton3,
-                   bidirectional>::processCellPairAoSN3(ParticleCell &cell1, ParticleCell &cell2,
-                                                        const std::array<double, 3> &sortingDirection) {
+template <class ParticleCell, class ParticleFunctor, bool bidirectional>
+void CellFunctor3B<ParticleCell, ParticleFunctor, bidirectional>::processCellPairAoSN3(
+    ParticleCell &cell1, ParticleCell &cell2, const std::array<double, 3> &sortingDirection) {
   if ((cell1.size() + cell2.size() > _sortingThreshold) and sortingDirection != std::array<double, 3>{0., 0., 0.}) {
```

Comment:
I'm hesitant to do so. I would imagine the compiler can do stuff with the constant value here.
And I think the {0,0,0} check is safe enough, otherwise something went wrong elsewhere.
(Also, this is the same in the normal `CellFunctor.h`)